sdv-mcp/tools.py

```
@@ -0,0 +1,188 @@
+
+        # Save synthetic data to CSV files
+        os.makedirs("synthetic_data", exist_ok=True)
+        for table_name, df in synthetic_data.items():
+            output_file = os.path.join("synthetic_data", f"{table_name}.csv")
+            df.to_csv(output_file, index=False)
+
+        return f"Data generated successfully and saved in 'synthetic_data' folder with {len(synthetic_data)} tables named as {list(synthetic_data.keys())} CSV files."
+
+    # Handle exceptions during data generation
+    except Exception as e:
+        raise RuntimeError(f"An error occurred while generating synthetic data: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)